skglm/experimental/pdcd_ws.py

```diff
@@ -88,10 +93,10 @@ def __init__(self, max_iter=1000, max_epochs=1000, dual_init=None,
         self.verbose = verbose
 
     def solve(self, X, y, datafit_, penalty_, w_init=None, Xw_init=None):
-        if issparse(X):
-            raise ValueError("Sparse matrices are not yet support in PDCD_WS solver.")
+        # jit compile classes
+        datafit = compiled_clone(datafit_)
```

Comment:
I think this is done only for this solver, is there a particular reason?

Reply:
Yes you are right.

I removed it. 
One thing is that is might have some side effects (for user) as the compilation was done in ``_validate_init``.